lib/nsync/initializer.coffee

```diff
@@ -197,12 +197,13 @@ module.exports = helper = (activationState) ->
 
   disposables.forEach (disposable) -> composite.add(disposable)
 
-  atomHelper.getToken().then (token) ->
-    nsync.configure
-      expansionState: activationState.directoryExpansionStates
-      localRoot: _path.join(atom.configDirPath, '.learn-ide')
-      connection:
-        url: "#{WS_SERVER_URL}?token=#{token}&version=#{atomHelper.learnIdeVersion()}"
+  atomHelper.waitForTerminalConnection().then ->
+    atomHelper.getToken().then (token) ->
+      nsync.configure
+        expansionState: activationState.directoryExpansionStates
+        localRoot: _path.join(atom.configDirPath, '.learn-ide')
+        connection:
+          url: "#{WS_SERVER_URL}?token=#{token}&version=#{atomHelper.learnIdeVersion()}"
```

Comment:
I actually ended up reworking this a bit to  handle errors better: https://github.com/learn-co/learn-ide-tree/pull/48/files#diff-23f01a445c8ded607cb8c67fbe954f90L200